### PR TITLE
Improve pointer ops

### PR DESCRIPTION
**Pull request by nirvdrum**

### CODE REVIEW

File: src/main/java/org/truffleruby/extra/ffi/Pointer.java [hunk 1]
```diff
@@ -90,21 +90,27 @@ public void writeBytes(long offset, long size, byte value) {
...
-        for (int n = 0; n < length; n++) {
-            writeByte(offset + n, bytes[index + n]);
-        }
+    public void writeBytes(long offset, byte[] buffer, int bufferPos, int length) {
+        assert address + offset != 0 || length == 0;
+        assert buffer != null;
+        assert bufferPos >= 0;
+        assert length >= 0;
+
+        UNSAFE.copyMemory(buffer, Unsafe.ARRAY_BYTE_BASE_OFFSET + bufferPos, null, address + offset, length);
```
Comment (eregon) [inline src/main/java/org/truffleruby/extra/ffi/Pointer.java]: Is it faster than the loop? By how much?

Comment (nirvdrum) [reply src/main/java/org/truffleruby/extra/ffi/Pointer.java]: Not massively. For the revcomp benchmark, it shaves ~7% off the time for `readlines`. But it's much more pronounced in the native image where there's a higher penalty for function calls.

**Comment by chrisseaton** [comment]
I don't really like these slow-path NFI function calls. Are they designed to be called like this?

File: src/main/java/org/truffleruby/extra/ffi/Pointer.java (hunk 1 above)
Comment (nirvdrum) [reply src/main/java/org/truffleruby/extra/ffi/Pointer.java]: I should note I haven't extensively benchmarked. I've really only looked at the one from #792. I believe `Unsafe` has optimized copy operations for aligned reads & writes.

**Comment by eregon** [comment]
> I don't really like these slow-path NFI function calls. Are they designed to be called like this?

The only thing not fast-path is that the EXECUTE Node is not adopted.
I'm not sure whether it matters or not.
But this is indeed the first fast-path usage of NativeFunction.

**Pull request merged**